app/workers/vic/delete_old_uploads.rb

```diff
@@ -1,82 +1,23 @@
 # frozen_string_literal: true
 
 module VIC
-  class DeleteOldUploads
-    include Sidekiq::Worker
+  class DeleteOldUploads < DeleteAttachmentJob
+    ATTACHMENT_CLASSES = %w[VIC::ProfilePhotoAttachment VIC::SupportingDocumentationAttachment].freeze
+    FORM_ID = 'VIC'
 
-    sidekiq_options(unique_for: 30.minutes, retry: false)
+    def get_uuids(form_data)
+      uuids = []
 
-    def perform
-      VIC::TagSentry.tag_sentry
-      delete_docs
-      delete_photos
-    end
-
-    def photos_to_keep
-      guids = []
-      ::InProgressForm.where(form_id: 'VIC').find_each do |ipf|
-        photo = ipf.data_and_metadata[:form_data]['photo']
-        guids << photo['confirmationCode'] if photo.present?
-      end
-
-      keep = []
-      ::VIC::ProfilePhotoAttachment.where(guid: guids).find_each do |photo|
-        keep << photo.get_file.path
-      end
-
-      keep
-    end
-
-    def docs_to_keep
-      guids = []
-      ::InProgressForm.where(form_id: 'VIC').find_each do |ipf|
-        docs = ipf.data_and_metadata[:form_data]['dd214']
-        docs.each { |doc| guids << doc['confirmationCode'] } if docs.present?
-      end
-
-      keep = []
-      ::VIC::SupportingDocumentationAttachment.where(guid: guids).find_each do |doc|
-        keep << doc.get_file.path
-      end
+      attachments = Array.wrap(form_data['photo'])
+      attachments += form_data['dd214'] || []
 
-      keep
-    end
-
-    private
-
-    # :nocov:
-    def delete_photos
-      keep = photos_to_keep
-
-      if Rails.env.production?
-        bucket.objects.with_prefix('profile_photo_attachments').delete_if do |obj|
-          obj.last_modified < 2.months.ago && keep.exclude?(obj.key)
+      if attachments.present?
+        attachments.each do |attachment|
```

Comment:
Nitpicky:  I think we can combine lines 11, 12 into `attachments = Array.wrap(form_data['photo']) + Array.wrap(form_data['dd214'])`, then remove the conditional on line 14 because we will always have an array to iterate through on line 15.